File: backend/models/memory.js
const mongoose = require('mongoose');

const memorySchema = new mongoose.Schema({
  title: String,
  description: String,
  eventDate: Date,
  tags: [String],
  image: String,
  lastReminderSent: Date,
});

const Memory = mongoose.model('Memory', memorySchema);
module.exports = Memory;

File: frontend/teacheron/lib/models/memory.dart
class Memory {
  final String id;
  final String title;
  final String description;
  final DateTime eventDate;
  final List<String> tags;
  final String image;
  final DateTime lastReminderSent;

  Memory({
    required this.id,
    required this.title,
    required this.description,
    required this.eventDate,
    required this.tags,
    required this.image,
    required this.lastReminderSent,
  });

  factory Memory.fromJson(Map<String, dynamic> json) {
    return Memory(
      id: json['_id'],
      title: json['title'],
      description: json['description'],
      eventDate: DateTime.parse(json['eventDate']),
      tags: List<String>.from(json['tags']),
      image: json['image'],
      lastReminderSent: DateTime.parse(json['lastReminderSent']),
    );
  }
}
